app/src/main/java/org/dicekeys/app/fragments/ScanFragment.kt

```diff
@@ -0,0 +1,185 @@
+    }
+
+    private lateinit var cameraProviderFuture: ListenableFuture<ProcessCameraProvider>
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            ActivityCompat.requestPermissions(requireActivity(), REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS)
+        }
+    }
+
+    // This is an arbitrary number we are using to keep track of the permission
+    // request. Where an app has multiple context for requesting permission,
+    // this can help differentiate the different contexts.
+    private val REQUEST_CODE_PERMISSIONS = 10
+
+    // This is an array of all the permission specified in the manifest.
+    private val REQUIRED_PERMISSIONS = arrayOf(Manifest.permission.CAMERA)
+
+    // Add this after onCreate
+    private val executor = Executors.newSingleThreadExecutor()
+
+
+//    private lateinit var previewView: PreviewView
+//    private lateinit var imageView: ImageView
+
+
+    override fun onRequestPermissionsResult(
+            requestCode: Int,
+            permissions: Array<out String>,
+            grantResults: IntArray
+    ) {
+        if (requestCode == REQUEST_CODE_PERMISSIONS) {
+            if (allPermissionsGranted()) {
+               binding.overlayView.post { startCamera() }
+            } else {
+                toast("Permissions not granted by the user.")
+                findNavController().popBackStack()
+            }
+        }
+    }
+
+    /**
+     * Check if all permission specified in the manifest have been granted
+     */
+    private fun allPermissionsGranted() = REQUIRED_PERMISSIONS.all {
+        ContextCompat.checkSelfPermission(requireContext(), it) == PackageManager.PERMISSION_GRANTED
+    }
+
+    private fun startCamera() {
+        cameraProviderFuture.addListener(Runnable {
+            val cameraProvider = cameraProviderFuture.get()
+            bindPreview(cameraProvider)
+        }, ContextCompat.getMainExecutor(requireContext()))
+    }
+
+    private fun bindPreview(cameraProvider: ProcessCameraProvider) {
+        var cameraSelector: CameraSelector = CameraSelector.Builder()
+                .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                .build()
+        //val previewSize = Size(viewFinder.width, viewFinder.height)
+        val previewSize = Size(binding.previewView.width, binding.previewView.height)
+
+
+        // Build the viewfinder use case
+        val preview: Preview = Preview.Builder()
+                .setTargetResolution(previewSize)
+                .build()
+
+        val pWidth = binding.previewView.width
+        val pHeight = binding.previewView.height
+        val analyzerSize: Size =
```

Comment:
1024x1024 should be enough to scan.  The higher the resolution, the slower the algorithm.

We want to analyze a square and show the user a square because the object we're scanning is square and we waste computation if scanning empty edges.

Reply:
This needs to be a separate issue. Had some issues with 1024x1024 so it needs more attention.